ember-toucan-core/src/components/form/textarea-field.ts

```diff
@@ -0,0 +1,30 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+
+import type { ToucanFormTextareaControlComponentSignature } from './controls/textarea';
+
+export interface ToucanFormTextareaFieldComponentSignature {
+  Element: HTMLTextAreaElement;
+  Args: {
+    error?: string;
+    hint?: string;
+    isDisabled?: boolean;
+    label: string;
+    onChange?: ToucanFormTextareaControlComponentSignature['Args']['onChange'];
```

Comment:
I was waffling back and forth on doing something like this in `controls/textarea`:

```js
export interface ToucanFormTextareaControlArguments {
   isDisabled?: boolean;
   onChange?: (value: string, e: Event | InputEvent) => void;
   value?: string;
 }
```

and then:

```js
onChange?: ToucanFormTextareaControlArguments['onChange'];
```

but it felt like an extra interface for not much gain.  Due to that, I opted for this as it's valid TS as well.  I have 0 strong opinions here though so will gladly do whatever folks think is best.

(My revert from above to this 😂 : https://github.com/CrowdStrike/ember-toucan-core/pull/36/commits/dc7902aff9d5429c288125a491b71e4abd71e0dd#diff-55352b7dd59af1213309f0921bd79cc8712b6d051513d50f3f23e3209ed52f11L4)

Reply:
I think this will be a common function signature, which e.g. the input component (control and -field) would also  need, right? In which case we could define it in a shared `types.ts`, something like:

```ts
type OnChangeCallback = (value: string, e: Event | InputEvent) => void;
// or, if we want to support values other than strings:
type OnChangeCallback<T> = (value: T, e: Event | InputEvent) => void;
```

I have similar shared types here: https://github.com/CrowdStrike/ember-headless-form/blob/main/packages/ember-headless-form/src/-private/types.ts#L35-L38

Reply:
Love this idea.  Went ahead and added it to a shared `types.ts` as suggested!